### PR TITLE
Add fft2d

### PR DESCRIPTION
**Pull request by chaithyagr**
This is a quick small PR to just add ortho FFT which is later used downstream by joint_optimization_mri, to get kspace from images. The idea here is to reduce burden for making multicoil data to single coil.

### CODE REVIEW

File: tf_fastmri_data/preprocessing_utils/fourier/cartesian.py [hunk 1]
```diff
@@ -30,3 +30,32 @@ def ortho_ifft2d(kspace):
...
     return scaling_norm * image
+
+
+def ortho_fft2d(image):
+    image = tf.cast(image, 'complex64')
+    axes = [len(image.shape) - 2, len(image.shape) - 1]
+    scaling_norm = tf.cast(tf.math.sqrt(tf.cast(tf.math.reduce_prod(tf.shape(image)[-2:]), 'float32')), image.dtype)
+    if len(image.shape) == 4:
+        # multicoil case
+        ncoils = tf.shape(kspace)[1]
```
Comment (zaccharieramzi) [inline tf_fastmri_data/preprocessing_utils/fourier/cartesian.py:41]: typo here
```suggestion
        ncoils = tf.shape(image)[1]
```

File: tf_fastmri_data/preprocessing_utils/fourier/cartesian.py [hunk 2]
```diff
@@ -30,3 +30,32 @@ def ortho_ifft2d(kspace):
...
+    n_slices = tf.shape(image)[0]
+    k_shape_x = tf.shape(image)[-2]
+    k_shape_y = tf.shape(image)[-1]
```
Comment (zaccharieramzi) [inline tf_fastmri_data/preprocessing_utils/fourier/cartesian.py:44]:
```suggestion
    i_shape_x = tf.shape(image)[-2]
    i_shape_y = tf.shape(image)[-1]
```

File: tf_fastmri_data/preprocessing_utils/fourier/cartesian.py [hunk 3]
```diff
@@ -30,3 +30,32 @@ def ortho_ifft2d(kspace):
...
+    shifted_image = fftshift(image, axes=axes)
+    batched_shifted_image = tf.reshape(shifted_image, (-1, k_shape_x, k_shape_y))
```
Comment (zaccharieramzi) [inline tf_fastmri_data/preprocessing_utils/fourier/cartesian.py:46]:
```suggestion
    batched_shifted_image = tf.reshape(shifted_image, (-1, i_shape_x, i_shape_y))
```

File: tf_fastmri_data/preprocessing_utils/fourier/cartesian.py [hunk 4]
```diff
@@ -30,3 +30,32 @@ def ortho_ifft2d(kspace):
...
+    batched_shifted_kspace = tf.map_fn(
+        fft2d,
+        batched_shifted_image,
+        parallel_iterations=multiprocessing.cpu_count(),
+    )
+    if len(image.shape) == 4:
+        # multicoil case
+        kspace_shape = [n_slices, ncoils, k_shape_x, k_shape_y]
```
Comment (zaccharieramzi) [inline tf_fastmri_data/preprocessing_utils/fourier/cartesian.py:54]:
```suggestion
        kspace_shape = [n_slices, ncoils, i_shape_x, i_shape_y]
```

File: tf_fastmri_data/preprocessing_utils/fourier/cartesian.py [hunk 5]
```diff
@@ -30,3 +30,32 @@ def ortho_ifft2d(kspace):
...
+    elif len(image.shape) == 3:
+        kspace_shape = [n_slices, k_shape_x, k_shape_y]
```
Comment (zaccharieramzi) [inline tf_fastmri_data/preprocessing_utils/fourier/cartesian.py:56]:
```suggestion
        kspace_shape = [n_slices, i_shape_x, i_shape_y]
```

File: tf_fastmri_data/preprocessing_utils/fourier/cartesian.py [hunk 6]
```diff
@@ -30,3 +30,32 @@ def ortho_ifft2d(kspace):
...
+    else:
+        kspace_shape = [k_shape_x, k_shape_y]
```
Comment (zaccharieramzi) [inline tf_fastmri_data/preprocessing_utils/fourier/cartesian.py:58]:
```suggestion
        kspace_shape = [i_shape_x, i_shape_y]
```

File: tf_fastmri_data/preprocessing_utils/fourier/cartesian.py [hunk 7]
```diff
@@ -30,3 +30,32 @@ def ortho_ifft2d(kspace):
...
+    shifted_kspace = tf.reshape(batched_shifted_kspace, kspace_shape)
+    kspace = fftshift(shifted_kspace, axes=axes)
```
Comment (zaccharieramzi) [inline tf_fastmri_data/preprocessing_utils/fourier/cartesian.py:60]:
```suggestion
    kspace = ifftshift(shifted_kspace, axes=axes)
```

File: tf_fastmri_data/preprocessing_utils/fourier/cartesian.py [hunk 8]
```diff
@@ -30,3 +30,32 @@ def ortho_ifft2d(kspace):
...
+    return scaling_norm * kspace
```
Comment (zaccharieramzi) [inline tf_fastmri_data/preprocessing_utils/fourier/cartesian.py:61]:
```suggestion
    return kspace / scaling_norm
```

**Review by zaccharieramzi — CHANGES_REQUESTED** [review]
Some slight mishaps in translation

**Comment by chaithyagr** [comment]
Arrggg, I was blindly copying codes from one branch to other, I think I missed something here..

**Comment by chaithyagr** [comment]
Oh no, i seem to have missed this codes. I dont know whats running currently then on Networks, the scaling is off in current codes xD
But i know for sure i did fix this, cause i did have scaling issues..

**Comment by chaithyagr** [comment]
> Oh no, i seem to have missed this codes. I dont know whats running currently then on Networks, the scaling is off in current codes xD
> But i know for sure i did fix this, cause i did have scaling issues..

Phew somehow JZ has the right codes.. I have zero clue how this happened. Phew.. XD

**Comment by zaccharieramzi** [comment]
@chaithyagr cool !
Could you also increase the version number by 0.1.0 (I forgot to ask for the addition of slices, this will do for the 2)?

**Comment by zaccharieramzi** [comment]
Otherwise LGTM, waiting for the change in version n0 and the greeen from tests

**Comment by chaithyagr** [comment]
> @chaithyagr cool !
> Could you also increase the version number by 0.1.0 (I forgot to ask for the addition of slices, this will do for the 2)?

I have some PRs affecting noisy codes, I was hoping to do them also maybe? Esp given that current codes will have issues in loaded data from what i remember for noisy data

**Comment by zaccharieramzi** [comment]
Let's do it for each (normally we should do it for each)

File: setup.py [hunk 9]
```diff
@@ -9,7 +9,7 @@
 
 setuptools.setup(
     name="tf-fastmri-data",
-    version="0.0.2",
+    version="0.2.0",
```
Comment (zaccharieramzi) [inline setup.py:12]: ha it was this low haha...
then maybe
```suggestion
    version="0.0.3",
```

**Pull request merged**